### DISCUSSION TITLE
Using AUX Port on Radiomaster MT12 for S.BUS Trainer

### DISCUSSION THREAD

**Discussion by armin-rc** [EdgeTX/edgetx#5311] [General]
Hello,

is the AUX port on the Radiomaster MT12 with MPM (4-in1) supposed to support Trainer via S.BUS? The corresponding setting in the menu is there (SYS > Hardware > Serial Port > AUX1 > SBUS Trainer) and in the model menu the setting is also set (MDL > SETUP > Trainer Mode > Master/SBUS). Of course I activated the trainer mode via a special function.

The whole setup is as follows:
The Rx (Radiolink R7FG) is connected via S.BUS to the AUX Port of the MT12. The blue LED on the Rx is constant on which indicates that it is running in S.BUS mode. A Radiomaster TX16S (which acts as the studend radio) is bound to that Rx. The MT12 is bound to another Rx in the model.

I red in a pull request on github that the UART in the MT12 cannot handle a "true" S.BUS signal (with "true" I mean as it was specified by Futaba [idle low / active high]) and that the signal has to be "inverted" (active low / idle high). I tried both signal variants and so far had no success. I checked the signals via an oscilloscope and a logic analyzer. They were looking good.

Does anybody know if it is supposed to work or am I hunting down a problem that cannot be resolved currently?

Link to the pull request:
https://github.com/EdgeTX/edgetx/pull/4668

Thanks,
Armin
arminRC

**Comment by tarantulito** [comment]
Hi.

Related to above, I have a Radiomaster MT12 ELRS LBT and can't do anything with the AUX Port, which has three pins on a Futaba type slot with a notch (S/+/-):
![MT12_AUXPort](https://github.com/user-attachments/assets/f10b9f2f-28c7-46b8-88bc-adf4abf9b154)

'+' has 5V permanently. I can't find 'S' description (when is it active, as input, as output...?), or LUA examples about its use. Other radios have 4pins ports, clearly labeled +/-/TX/RX.

I am trying this:
<img width="940" height="530" alt="image" src="https://github.com/user-attachments/assets/0c02c1d2-f3ab-4b26-934d-f8878e4c4e35" />
and trying serial output on the 'S' pin (which could work if 'S' (_LUA_) were output in this case) with a simple LUA script:
```
local BAUDRATE = 115200
local STRINGA = "Hola, mundo"
local counter = 0

local function init()
counter = 0
    setSerialBaudrate(BAUDRATE)
    lcd.clear()
    lcd.drawText(5, 5, "[AUX Serial Test] Iniciado a " .. BAUDRATE .. " baudios")
    return 1
end

local lastSent = 0
local interval = 50
local exited = false

local function run(event)
  if exited then
    return 1
  end
  local now = getTime()
  if now - lastSent > interval then
    serialWrite(STRINGA)
    counter = counter + 1
    lastSent = now
  end
  lcd.clear()
  lcd.drawText(5, 5, STRINGA .. counter)
  if event == EVT_VIRTUAL_EXIT then
    lcd.clear()
    lcd.drawText(5, 5, "Script exit executed")
    exited = true
    return 1
  end
  return 0
end

return { init=init, run=run }
```
`counter` is seen incremented, but the 'S' pin is always at 0 level (as seen on oscilloscope).
`setSerialBaudrate(BAUDRATE)` and `serialWrite(STRINGA)` agree to [documentation](https://luadoc.edgetx.org/lua-api-reference/serial-ports).

If trying with the USB port (_Telem Mirror_) there appear the CRSF frames, which I can read with Python.

**Any clue about the 'S' pin?**

**Comment by DogRC** [comment]
I know it's quite old but is there any update on this?

I'm experiencing the same issue, but have also tried using the exposed pins of the top module bay and not having any luck.

Have tried various combinations of settings.

Originally using version 2.10.0 of EdgeTX, but later upgraded to 2.12.1 with no change.
